subprojects/version-control/src/integTest/groovy/org/gradle/vcs/git/internal/VcsCleanupIntegrationTest.groovy

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.vcs.git.internal
+
+import org.eclipse.jgit.revwalk.RevCommit
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.internal.hash.HashUtil
+import org.gradle.test.fixtures.file.TestFile
+import org.gradle.vcs.fixtures.GitRepository
+import org.junit.Rule
+
+import java.util.concurrent.TimeUnit
+
+class VcsCleanupIntegrationTest extends AbstractIntegrationSpec {
```

Comment:
Are there any scenarios where we don't expect cache cleanup to occur?  For instance, when dependencies are not resolved or when no vcs mappings are defined.

If not, I wonder if we should have at least one test that validates that caches are cleaned up even when all vcs mappings are removed and no dependencies are resolved.

Reply:
This is a good idea, i'll do that.